eva-accession-import-automation/validate_imported_contigs_for_assembly.py

```diff
@@ -0,0 +1,132 @@
+        errors = os.linesep.join(process.stderr.readlines())
+    if process.returncode != 0:
+        logger.error(command_description + " failed!" + os.linesep + errors)
+        raise subprocess.CalledProcessError(process.returncode, process.args)
+    else:
+        logger.info(command_description + " completed successfully")
+    return process.returncode, process_output
+
+
+def get_args_from_assembly_properties_file(assembly_properties_file):
+    parser = configparser.ConfigParser()
+    parser.optionxform = str
+
+    with open(assembly_properties_file, "r") as properties_file_handle:
+        # Dummy section is needed because
+        # ConfigParser is not clever enough to read config files without section headers
+        properties_section_name = "assembly_properties"
+        properties_string = '[{0}]\n{1}'.format(properties_section_name, properties_file_handle.read())
+        parser.read_string(properties_string)
+        config = dict(parser.items(section=properties_section_name))
+        return config
+
+
+def validate_imported_contigs(assembly_properties_file, config_file):
+    config = get_args_from_private_config_file(config_file)
+    config.update(get_args_from_assembly_properties_file(assembly_properties_file))
+    config["assembly_report_path"] = config["parameters.assemblyReportUrl"].split("file:")[-1]
+    config["assembly_md5"] = hashlib.md5(config["parameters.assemblyName"].encode("utf-8")).hexdigest()
+    # We need these because string interpolation won't work if there is a dot in the key
+    config["taxonomy_accession"] = config["parameters.taxonomyAccession"]
+
+
+    config["contig_chr_mismatch_table"] = "dbsnp_ensembl_species.dbsnp_species_with_contig_chromosome_start_mismatch"
+    config["contig_chr_match_table"] = "dbsnp_ensembl_species.dbsnp_species_with_contig_chromosome_start_match"
+    final_formatting_cmd = r"cut -d$'\t' -f5 | sort | uniq | paste -s - | sed s/$'\t'/\",\"/g | sed s/$/\"/g | sed s/^/\"/g"
+    get_contigs_start_mismatch_cmd = "psql -A -t -h {metahost} -U {metauser} -d {metadb} -c " \
+                                     "\"select distinct contig_name from {contig_chr_mismatch_table} " \
+                                     "where table_name like '%{assembly_md5}%'" \
+                                     " and schema_name like '%{taxonomy_accession}%'\" -P pager=off " \
+                                     "| grep {assembly_report_path} -f - |".format(**config) + final_formatting_cmd
+    get_contigs_start_match_cmd = "psql -A -t -h {metahost} -U {metauser} -d {metadb} -c " \
+                                  "\"select distinct contig_name from {contig_chr_match_table} " \
+                                  "where table_name like '%{assembly_md5}%'" \
+                                  " and schema_name like '%{taxonomy_accession}%'\" -P pager=off " \
+                                  "| grep {assembly_report_path} -f - | grep -v assembled-molecule |".format(**config) \
```

Comment:
why removing the assembled-molecules?

Reply:
We are taking out the assembled-molecule lines because we are looking for contigs where contig_start = chr_start but are not actually in the assembled-molecule entries. 

Given that RS2 is a RefSeq contig in chr1 in dbSNP with **contig_start = chr_start** and the following in the assembly report:
chr1 assembled-molecule CM1 = RS1
RS2  scaffold NONCM1 = RS2 

The candidate that we should look for is RS2 but not RS1 because RS1 would have definitely been replaced with CM1 by the older version of the pipeline but RS2 won't be.

Reply:
oh, I see, this is why sheep and another species give false positives. ok